my-ansible-managed-playbook.yml

```diff
@@ -0,0 +1,226 @@
+
+- name: Connect to managed nodes
+  hosts: my-managed-nodes
+  remote_user: ec2-user
+
+  vars_prompt:
+    - name: flask_env
+      prompt: What is the flask_env?
+      private: no
+      default: development
+
+    - name: secret_key
+      prompt: What is your secret key?
+      default: secret_key
+
+    - name: trello_token
+      prompt: What is your trello api token?
+      default: trello_token
+    
+    - name: user_name
+      prompt: What is your user name?
+      private: no
+      default: todoapp_user
+
+  tasks:
+  - name: Create user {{ user_name }} in ~{{ user_name }}
+    ansible.builtin.user:
+      name:  "{{ user_name }}"
+      comment: Todoapp user - {{ user_name }}
+    become: yes
+    
+  - name : Install Git and Python3 on Managed Node
+    ansible.builtin.yum:
+      name: 
+        - git
+        - python3
+      state: latest
+    register: git_python_install
+    become: yes
+
+  - debug:  
+      var: git_python_install.stdout_lines
+  
+  - name : Capture Git version on Managed Node
+    ansible.builtin.shell: git version
+    register: git_version
+  
+  - debug:  
+      var: git_version.stdout_lines
+
+  - name : Capture Python3 version on Managed Node
+    ansible.builtin.shell: python3 --version
+    register: py_version
+  
+  - debug:  
+      var: py_version.stdout_lines
+
+  - name : Capture Curl version on Managed Node
+    ansible.builtin.shell: curl --version
+    register: curl_version
+  
+  - debug:  
+      var: curl_version.stdout_lines
+
+  - name: Install Poetry
+    ansible.builtin.shell: curl -sSL https://install.python-poetry.org | python3 -
+    args: 
+      creates: ~/.local/bin/poetry
+    register: poetry_install
+    become: yes
+    become_user: "{{ user_name }}"
+
+  - debug:  
+      var: poetry_install.stdout_lines
+
+  - name : Capture Poetry version on Managed Node
+    ansible.builtin.shell: which poetry && poetry --version
+    register: poetry_version
+    environment:
+      PATH: "~/.local/bin:{{ ansible_env.PATH }}"
+    become: yes
+    become_user: "{{ user_name }}"
+  
+  - debug:  
+      var: poetry_version.stdout_lines
+
+  - name: Create todoapp directory
+    ansible.builtin.file:
+      path: /opt/todoapp
+      state: directory
+      mode: '477'
+    become: yes
+
+  - name : Make todoapp a git safe directory on Managed Node
+    ansible.builtin.shell: git config --global --add safe.directory /opt/todoapp
+    register: todoapp_safedir
+  
+  - debug:  
+      var: todoapp_safedir.stdout_lines
+
+  - name: Checkout latest version of todoapp
+    ansible.builtin.git:
+      repo: 'https://github.com/dagoud/DevOps-Course-Starter.git' 
+      dest: /opt/todoapp
+      version: module-4/exercise
+    register: todoapp_git_checkout
+    become: yes
+
+  - debug:  
+      var: todoapp_git_checkout.stdout_lines
+
+  - name : Change dir to todoapp on Managed Node
+    ansible.builtin.shell: git status
```

Comment:
I don't think the name matches the action here

Reply:
Updated thanks